core/src/utils/test/playwright/drag-element.ts

```diff
@@ -55,12 +67,76 @@ export const dragElementByYAxis = async (
   const startX = boundingBox.x + boundingBox.width / 2;
   const startY = startYCoord === undefined ? boundingBox.y + boundingBox.height / 2 : startYCoord;
 
+  // Navigate to the start position.
   await page.mouse.move(startX, startY);
   await page.mouse.down();
 
-  for (let i = 0; i < dragByY; i += 20) {
-    await page.mouse.move(startX, startY + i);
-  }
+  // Drag the element.
+  await moveElement(page, startX, startY, 0, dragByY);
 
   await page.mouse.up();
 };
+
+const calculateEndX = (startX: number, dragByX: number, viewportWidth: number) => {
```

Comment:
I wonder if this should error instead of trying to adjust the values in-bounds. My concern is that if we adjust the values such that certain gestures no longer work as expected, this may be hard to debug if team members don't know their values are being adjusted for them.

Reply:
Alternatively, is there info we can expose that lets us set safe values without having to set explicit pixel values? For example, I might want to have a gesture that pulls all the way down to the bottom of the viewport. I don't care what the exact pixel coordinate is, as long as it is the bottom of the viewport.

Reply:
How about `dragByX: number | undefined`? If undefined then we would drag all the way to the bottom. The same would go for `calculateEndY`. If we go this route, then the parameter `dragByX` from `dragElementBy` should update to also handle undefined.

Reply:
I assume using a Direction would drag to the far edge of the element's bounding box?

Also the literal up/down language is a little confusing when applied to the X axis, but admittedly I can't think of a better way of capturing it 😆 My immediate thought is having `dragElementBy` accept `from` and `to` params that can be compass directions (e.g. `from='center'`, `to='northeast'`) but that might be overkill.

Reply:
It might be good to do some design work on this. I do think there's a nice solution in here somewhere, I just can't think of it at the moment. How do we feel about erroring/warning when the coordinate is out of bounds for now and letting the test implementer make the adjustment themselves? That way it's clear to the implementer what is happening rather than the coordinates being adjusted without them knowing.

Reply:
Understood, I'll update the code to error for now and create a design doc for a better solution. Once the design doc gets approved, I'll make a new PR to implement the solution.